src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -147,21 +147,28 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 	private registerGroupListeners(group: IEditorGroupView): void {
 		const groupDisposables = new DisposableStore();
 
-		groupDisposables.add(group.onDidGroupChange(e => {
+		groupDisposables.add(group.onDidModelChange(e => {
```

Comment:
Here I find it somewhat dangerous to suddenly change away from `onDidGroupChange` to the `onDidModelChange`. Maybe the UI component decides to fire events differently compared to the model. 

I would suggest to expose a new event from editor service to subscribe to any model event and then use that from the tabs API.

Reply:
This is only used by the tabs API. You can see below I have a separate one for UI events which is needed for editor activity

Reply:
Ok.

Reply:
Why exactly do we distinguish the `GroupChangeKind.EDITOR_ACTIVE` here from the default case?

Reply:
The event fires to all groups even if they aren't active. We really only care about the active group's active editor. I can probably move that logic into the tab code though

Reply:
Yes please do.